ql/src/java/org/apache/hadoop/hive/ql/security/authorization/command/CommandAuthorizerV2.java

```diff
@@ -139,6 +139,13 @@ private static List<HivePrivilegeObject> getHivePrivObjects(List<? extends Entit
         continue;
       }
 
+      if (privObject.getTyp() == Type.FUNCTION && !HiveConf.getBoolVar(SessionState.get().getConf(),
+              HiveConf.ConfVars.HIVE_AUTHORIZATION_FUNCTIONS_IN_VIEW) && hiveOpType == HiveOperationType.QUERY) {
+        if (!sem.getUserSuppliedFunctions().contains(privObject.getFunctionName())) {
+          continue;
```

Comment:
Is it possible to add test cases that show the behavior in a couple of cases (maybe similar to the authorization_view_*.q tests?). Ideally - I would like to see a query with a user supplied udf with a table, a view, and see how it also interacts with MVs. Ideally the q.out file would show which objects are being sent for authorization so it can catch the regression on when it sends an extra UDFs for auth (or doesn't send the user supplied one for auth).

Reply:
Ack